query/physicalplan/physicalplan.go

```diff
@@ -260,17 +333,21 @@ func Build(ctx context.Context, pool memory.Allocator, tracer trace.Tracer, s *d
 			return false
 		}
 
-		for i := 0; i < len(phyPlans); i++ {
-			phyPlans[i].SetNext(prev[i])
+		// If these aren't the same the operator should handle how to synchronize
+		if len(phyPlans) == len(prev) {
+			for i := 0; i < len(phyPlans); i++ {
+				phyPlans[i].SetNext(prev[i])
+			}
 		}
+
 		prev = phyPlans
 		return true
 	}))
 
 	if len(prev) == 1 {
 		outputPlan.SetNextCallback(prev[0].Callback)
 	} else {
-		// TODO: Update to use sychronizer's single callback if we have concurrency.
+		// This case should be handled in the concurrent operator above.
```

Comment:
Maybe returning errors here would be good. Or maybe a panic since this would be programming error?